### ISSUE TITLE
Editor doesn't preserve image size when pasting resized images from Google Docs

### ISSUE THREAD

**Issue by jswiderski** [ckeditor/ckeditor5#10838]
## 📝 Provide detailed reproduction steps (if any)

1.  Create sample document in Google Docs with resized image.
2.  Copy the image and paste it into editor.

### ✔️ Expected result

Pasted images has the same size as in Google Docs.

### ❌ Actual result

Image is pasted with its original size.

Image HTML has width/height attributes which doesn't seem to be taken into account when pasting:

```
<img src="https://lh4.googleusercontent.com/APWThOWTll8I0A_cIYAEe-o69Ddbc-uul0YkfwnSVnhZp9GoRvtCENXG50sq6sx9A0nKMw7C64nju3IU0gyPd7Xg6YxfG2Ul3hn_nqsj7jEUoAXbMEaNERGVkagEa57zooZWr-6L" width="161" height="226" style="margin-left:0px;margin-top:0px;" />
```

### ❓ Possible solution

_If you have ideas, you can list them here. Otherwise, you can delete this section._

## 📃 Other details

*   Browser: Any
*   OS: Any
*   First affected CKEditor version: N/A
*   Installed CKEditor plugins: Image

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by jswiderski** [comment]
Similar or related issues - #5632, #5154

**Comment by jswiderski** [comment]
Even if you set GHS to allow width/height attributes, it will not have any effect:

```
htmlSupport: {
			disallow: [
				{
					name: 'img',
					styles: {
						width: true,
						height: true
					}
				}
			],	
			allow: [
				{
					name: 'img',
					attributes: {
						width: true,
						height: true
						
					}
				}
			]
}
```

Attributes are assigned to wrapping `span` but not to `img` element.

![](https://user-images.githubusercontent.com/5314687/144839187-cf7a7009-e670-4e5a-a384-a447c6be264c.png)

Google Docs:

![](https://user-images.githubusercontent.com/5314687/144839686-ce61db3a-a8f0-4fb2-be30-bda1b8b2cb0f.png)

**Comment by mlewand** [comment]
I did a quick assessment on this.

Google simply uses `img[height]` and `img[width]` attributes to convey image size. We don't have any conversion for this.

All we need to fix size retention for proportionally resized images is to add converter for `img[width]` attribute.

However, we don't have any handling for image height in CKE5 so far, so if we were to handle non-proportionally resized image we'd need to add a handling for that which is a bigger task.

**Comment by wimleers** [comment]
> Google simply uses `img[height]` and `img[width]` attributes to convey image size. We don't have any conversion for this.

Then I'm pretty sure this will also be a problem for Drupal.

**Comment by mmichaelis** [comment]
I think, this may also be a related issue:

* #11393

**Comment by Witoso** [comment]
This issue was solved with the #14146.
